src/main/java/net/openhft/chronicle/core/time/PosixTimeProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016-2020 chronicle.software
+ *
+ * https://chronicle.software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.openhft.chronicle.core.time;
+
+import net.openhft.posix.ClockId;
+import net.openhft.posix.PosixAPI;
+
+/**
+ * This is the raw Posix time, if you need it to be unique or a multiple of 32 use a wrapper.
+ */
+public enum PosixTimeProvider implements TimeProvider {
```

Comment:
what is the difference between this and Chronicle-Ticker?

Reply:
Good question, let be review it.

Reply:
I've added @minborg for review as I think he suggests not using enums for utility methods like this.

Reply:
I think we should have a utility class that provides these different time providers and keep the actual implementations in an internal package. 

For example, `TimeProviders.acquirePosixTimeProvider()` or `TimeProviders.createUniqueMicroTimeProvider()`

Reply:
`SetTimeProvider` is using a non-enum pattern which is better from an API perspective.

![image](https://user-images.githubusercontent.com/7457876/148062276-adbef03a-787c-4840-9306-f1851795f4b1.png)

Reply:
Are these classes to be used in conjunction with annotations? If not, we should really internalize them in the long run. We could move them to an internal package and deprecate the old ones (which could delegate form the internal ones and only act as a compatibility facade)

Reply:
I thought we agreed at a dev meeting long ago we were going to continue to use enums for our singletons? We need to be consistent as right now we have different people doing different things

Reply:
Enums for singletons are fine with me. But they are best put in an internal package and exposed by other means.